src/main/java/org/opensearch/knn/index/memory/NativeMemoryLoadStrategy.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.memory;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.knn.index.JNIService;
+import org.opensearch.knn.index.util.KNNEngine;
+import org.opensearch.knn.training.TrainingDataConsumer;
+import org.opensearch.knn.training.VectorReader;
+import org.opensearch.watcher.FileChangesListener;
+import org.opensearch.watcher.FileWatcher;
+import org.opensearch.watcher.ResourceWatcherService;
+import org.opensearch.watcher.WatcherHandle;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Responsible for loading entries from native memory.
+ */
+public interface NativeMemoryLoadStrategy<T extends NativeMemoryAllocation, U extends NativeMemoryEntryContext<T>> {
+
+    /**
+     * Loads a NativeMemoryAllocation from a given NativeMemoryEntryContext
+     *
+     * @param nativeMemoryEntryContext Context used to load nativeMemoryAllocation
+     * @return Loaded NativeMemoryAllocation
+     */
+    T load(U nativeMemoryEntryContext) throws IOException;
+
+    class IndexLoadStrategy implements NativeMemoryLoadStrategy<NativeMemoryAllocation.IndexAllocation,
+            NativeMemoryEntryContext.IndexEntryContext> {
+
+        public static IndexLoadStrategy INSTANCE = new IndexLoadStrategy();
+
+        private final FileChangesListener indexFileOnDeleteListener;
+        private ResourceWatcherService resourceWatcherService;
+
+        public static void initialize(final ResourceWatcherService resourceWatcherService) {
+            INSTANCE.resourceWatcherService = resourceWatcherService;
+        }
+
+        /**
+         * Empty Constructor
+         */
+        public IndexLoadStrategy() {
```

Comment:
why do we have public static and public constructor?

Reply:
Will change